time timestamp
int32 w1pos
int32 w2pos
